### DISCUSSION TITLE
Troubleshooting playback issues in YouTube and YouTube Music

### DISCUSSION THREAD

**Discussion by oSumAtrIX** [ReVanced/revanced-discussions#1037] [General]
# Setup

> [!IMPORTANT]
> ~~If you have recently started to experience playback issues, you may have been put in an A/B bucket by YouTube, for which the following troubleshooting steps will not work. The issue is currently being investigated.~~ Fixed

If you experience playback issues in YouTube and YouTube Music, ensure the following:

1. You patched the app with the `GmsCore support` patch using the latest version of [ReVanced Patches](https://github.com/ReVanced/revanced-patches)
1. [GmsCore](https://github.com/ReVanced/GmsCore) is installed (If you patched your app correctly, you should receive a notice about this when launching the app)
3. Your system is configured correctly to ignore battery optimizations for GmsCore, and GmsCore is allowed to run in the background (If you patched your app correctly, you should receive a notice about this when launching the app)
4. SafetyNet is enabled in the GmsCore settings
5. SafetyNet is correctly set up on your system. Pass SafetyNet checks using a SafetyNet checker such as [YASNAC - SafetyNet Checker](https://play.google.com/store/apps/details?id=rikka.safetynetchecker)

# Troubleshooting

If playback still fails, follow these steps:

1. Launch the patched app and ensure that playback fails
2. Close the patched app
3. Open a terminal (such as [Termux](https://f-droid.org/de/packages/com.termux/)) and run `logcat -c | logcat | grep PoToken` via an ADB shell (Research yourself how to do this step)
4. Launch the patched app

## Evaluation

When you started the patched apps, you should see logs appearing in your terminal.

- If you see `PoIntegrityTokenResponse { <A long Base64 string> }` or `responseStatusToken result: <Count of PoToken bytes>`, you should not experience any typical playback issues
- If `GetPoIntegrityTokenResponse{}` appears in the logs, DroidGuard did not respond with a `PoIntegrityToken`. Either SafetyNet is not enabled in the GmsCore settings, or DroidGuard/ SafetyNet is not correctly set up on your system.

## Report

Update: See https://github.com/ReVanced/revanced-discussions/discussions/1037#discussioncomment-9307345

If you experience playback issues despite following all the sections above, reply with a comment using the following template:

 ```md
## Environment

Android version:
Custom ROM: Yes (Name of ROM)/No/Don't know

## Screenshot of passing SafetyNet checks using a SafetyNet checker

<!-- Drag-drop your screenshot below this line -->

## Screenshot of GmsCore settings, especially the SafetyNet setting

<!-- Drag-drop your screenshot below this line -->

## Logs captured from the troubleshooting section

<!-- Add logs here -->
 ```

# Notice about moderation of this discussion

Comments that do not follow the comment template from above or are spam, duplicate, redundant, or abusive will be hidden or removed. Repeated offenders will be blocked.

**Comment by jeregon1** [comment]
## Environment

Android version: 14
Custom ROM: No

## Screenshot of passing SafetyNet checks using a SafetyNet checker

<img src="https://github.com/ReVanced/revanced-discussions/assets/131779521/db0e3c02-53b0-4e17-a9f5-51dc837c0cb1" alt="Screenshot of passing SafetyNet checks using a SafetyNet checker" width="450"/>

## Screenshot of GmsCore settings, especially the SafetyNet setting

Sorry it's in French, ask me anything
Auto-verification:

<img src="https://github.com/ReVanced/revanced-discussions/assets/131779521/f4b7ba4f-be37-4930-bb42-65f938cab8d4" alt="Screenshot of passing SafetyNet checks using a SafetyNet checker" width="450"/>

Terminal registration:
<img src="https://github.com/ReVanced/revanced-discussions/assets/131779521/2c839f1a-b724-481e-af12-faf4248c4d88" alt="Screenshot of passing SafetyNet checks using a SafetyNet checker" width="450"/>

Safety-net:
<img src="https://github.com/ReVanced/revanced-discussions/assets/131779521/b39e3574-d097-4e24-b39c-781ebd5844f1" alt="Screenshot of passing SafetyNet checks using a SafetyNet checker" width="450"/>


## Logs captured from the troubleshooting section

```
ruby:/ $ logcat -c | logcat | grep PoToken
04-05 10:21:37.254 20107 21486 D ChimeraServiceProvider: serviceIntentCall: app.revanced.android.gms.potokens.service.START -> Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
04-05 10:21:37.268 20107 20107 D PoTokens: onBind: Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
04-05 10:21:37.844 20107 20451 D PoTokens: bound by: GetServiceRequest{serviceId=POTOKENS, gmsVersion=241002000, packageName='app.revanced.android.youtube', extras=Bundle[{}]}
04-05 10:21:37.844 20107 20451 D PoTokens: PoTokensApiService handleServiceRequest
04-05 10:21:38.085 20107 21651 D PoTokens: responseStatusToken packageName: com.google.android.apps.youtube
04-05 10:21:38.086 20107 20107 D PoTokens: responseStatusToken start
04-05 10:21:38.094 20107 20107 D PoTokens: responseStatusToken result: 236
04-05 10:21:38.095 20107 20107 D PoTokens: responseStatusToken end
^C
```

Thank you so much guys!!!

**Reply by jeregon1** [reply]
UPDATE: after rebooting it WORKS!!

I'm in a Redmi Note 12 Pro 5G, only worked after enabling safety net and rebooting.
No battery optimization disabled for YouTube itself.
You can delete this, thanks for the support!

**Comment by GKid94** [comment]
Have the problem on Redmi 9, videos keep buffering and won't play
![Screenshot_2024-04-07-23-45-04-547_app revanced android youtube](https://github.com/ReVanced/revanced-discussions/assets/18757054/2cf27f52-a38e-43b1-9570-2ffffd40e537)

**Comment by GKid94** [comment]
![Screenshot_2024-04-08-00-04-16-497_rikka safetynetchecker](https://github.com/ReVanced/revanced-discussions/assets/18757054/6b65a563-3c6d-4cb2-b838-c9abd8bbc09c)
![Screenshot_2024-04-08-00-04-48-256_app revanced android gms](https://github.com/ReVanced/revanced-discussions/assets/18757054/d3fbd36d-c485-46a6-aabf-487c43c18edd)

**Comment by cyberboh** [comment]
@GKid94
Capture logs as of it needed to troubleshooting. Read 1st post properly.

**Comment by seadx6-neo** [comment]
I am having playback issues on my OPPO A17s, it plays a little and then stops
![Screenshot_2024-05-02-10-02-00-16_0ecf870e1d5390b9ac21ebc76f00749d](https://github.com/ReVanced/revanced-discussions/assets/131829238/841d41cb-9967-4284-9e0b-caa491f7106f)
![Screenshot_2024-05-02-10-08-29-76_64ef5fc2000c1caa954c114bb372e1d5](https://github.com/ReVanced/revanced-discussions/assets/131829238/c9ae4de3-f895-4635-9c60-690bad8a0e8d)

**Comment by OmegaSunkey** [comment]
## Environment

Android version: 11
Custom ROM: No

## Screenshot of passing SafetyNet checks using a SafetyNet checker

![Screenshot_20240502-204620_Yet Another SafetyNet Attestation Checker](https://github.com/ReVanced/revanced-discussions/assets/94485936/19681333-e1ef-4793-b5ae-6a22cbcaa772)


## Screenshot of GmsCore settings, especially the SafetyNet setting

![Screenshot_20240502-210201_microG Services](https://github.com/ReVanced/revanced-discussions/assets/94485936/1fe5d9c6-bbe6-4c5f-9b5c-2a7d44307cc8)
![Screenshot_20240502-205250_microG Services](https://github.com/ReVanced/revanced-discussions/assets/94485936/cf28195e-d7da-4a55-84f6-11d75f5083fe)


## Logs captured from the troubleshooting section

```
a30:/ $ logcat -c | logcat | grep PoToken
05-02 20:49:48.126 27908 27929 D ChimeraServiceProvider: serviceIntentCall: app.revanced.android.gms.potokens.service.START -> Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
05-02 20:49:48.144 27908 27908 D PoTokens: onBind: Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
05-02 20:49:48.165 27908 27929 D PoTokens: bound by: GetServiceRequest{serviceId=POTOKENS, gmsVersion=241199000, packageName='app.revanced.android.youtube', extras=Bundle[{}]}
05-02 20:49:48.165 27908 27929 D PoTokens: PoTokensApiService handleServiceRequest
05-02 20:49:49.199 27908 27929 D PoTokens: responseStatusToken packageName: com.google.android.apps.youtube
05-02 20:49:49.237 27908 27908 D PoTokens: responseStatusToken start
05-02 20:49:49.417 27908 27908 D PoTokens: responseStatusToken result: 244
05-02 20:49:49.420 27908 27908 W SafeParcel: class com.google.android.gms.potokens.PoToken has no default constructor
05-02 20:49:49.423 27908 27908 D PoTokens: responseStatusToken end
```


# More info

After some video playback, the stream stops entirely. 
![Screenshot_20240502-202859](https://github.com/ReVanced/revanced-discussions/assets/94485936/e00e40d9-4138-4618-a378-4de31f921f56)
![Screenshot_20240502-203241](https://github.com/ReVanced/revanced-discussions/assets/94485936/e752fbc5-97ba-43dd-9f95-7c070886d87f)
![Screenshot_20240502-203810](https://github.com/ReVanced/revanced-discussions/assets/94485936/1b32dde5-d620-4a98-a4c3-c46414b03d29)

You can see that it stops playback randomly. Not after 40 seconds, not after 1 minute, just any time it wants.

Maybe unrelated, but video startup has a noticeable delay. It takes longer depending on the duration of the video.

**Reply by oSumAtrIX** [reply]
Is the screenshot taken from the launch of the app that produced the log above?

**Reply by OmegaSunkey** [reply]
Nope, different timestamps.

**Reply by oSumAtrIX** [reply]
Check the playback in the same instance of the app that produced the log:

responseStatusToken result:  <byteCount>

**Reply by OmegaSunkey** [reply]
New session.
Log output: 
```
05-02 21:06:37.503  5141  5178 D ChimeraServiceProvider: serviceIntentCall: app.revanced.android.gms.potokens.service.START -> Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
05-02 21:06:37.521  5141  5141 D PoTokens: onBind: Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
05-02 21:06:37.529  5141  5178 D PoTokens: bound by: GetServiceRequest{serviceId=POTOKENS, gmsVersion=241199000, packageName='app.revanced.android.youtube', extras=Bundle[{}]}
05-02 21:06:37.529  5141  5178 D PoTokens: PoTokensApiService handleServiceRequest
05-02 21:06:38.272  5141  5178 D PoTokens: responseStatusToken packageName: com.google.android.apps.youtube
05-02 21:06:38.281  5141  5141 D PoTokens: responseStatusToken start
05-02 21:06:38.395  5141  5141 D PoTokens: responseStatusToken result: 244
05-02 21:06:38.396  5141  5141 W SafeParcel: class com.google.android.gms.potokens.PoToken has no default constructor
05-02 21:06:38.398  5141  5141 D PoTokens: responseStatusToken end
```

Videos still stop at a random timestamp.

**Reply by oSumAtrIX** [reply]
If the PoToken is generated and supplied to YouTube but playback still fails, then one reason could be that Google updated DroidGuard checks, so that the PoToken till doesn't allow playback. This would need further debugging. I'd need an account where this issue occurs though to debug this. Anyone having an affected account they can share can contact me at osumatrix@revanced.app for further instructions. No login credentials are needed.

**Reply by oSumAtrIX** [reply]
A quick search is showing signs of this being an issue with YouTube:

![Screenshot_20240503-042021](https://github.com/ReVanced/revanced-discussions/assets/13122796/d20d5668-c718-4121-8a88-922e6d7b4f80)

https://www.reddit.com/r/youtube/comments/1ciwrp2/_/

**Reply by OmegaSunkey** [reply]
Hopefully its this and not me winning the lottery

**Reply by wolfdael** [reply]
Clearing cache in yt and microg plays the video in 720p only. This is what works for me for now

**Comment by adsadownloads** [comment]
## Environment

Android version: 14
Custom ROM: No

## Screenshot of passing SafetyNet checks using a SafetyNet checker


![Screenshot_20240503_121258_Yet_Another_SafetyNet_Attestation_Checker](https://github.com/ReVanced/revanced-discussions/assets/158696316/cd16c886-d8c3-47a5-bbc7-e2659047cc4c)

## Screenshot of GmsCore settings, especially the SafetyNet setting


![Screenshot_20240503_121353_microG Services](https://github.com/ReVanced/revanced-discussions/assets/158696316/56fbdc8c-4061-4e66-81bd-81bbe012a35a)
![Screenshot_20240503_121307_microG Services](https://github.com/ReVanced/revanced-discussions/assets/158696316/20929253-c837-417d-99bf-e5659c737eab)
## Logs captured from the troubleshooting section

```
05-03 13:03:26.268  2894  6260 D ChimeraServiceProvider: serviceIntentCall: app.revanced.android.gms.potokens.service.START -> Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
05-03 13:03:26.272  2894  2894 D PoTokens: onBind: Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
05-03 13:03:26.758  2894  2909 D PoTokens: bound by: GetServiceRequest{serviceId=POTOKENS, gmsVersion=241199000, packageName='app.revanced.android.youtube', extras=Bundle[{}]}
05-03 13:03:26.758  2894  2909 D PoTokens: PoTokensApiService handleServiceRequest
05-03 13:03:26.941  2894  7127 D PoTokens: responseStatusToken packageName: com.google.android.apps.youtube
05-03 13:03:26.970  2894  2894 D PoTokens: responseStatusToken start
05-03 13:03:26.978  2894  2894 D PoTokens: responseStatusToken result: 243
05-03 13:03:26.979  2894  2894 D PoTokens: responseStatusToken end
```

## More info
[EDIT 1&2&3:] It's either the playback doesn't start, or starts then stops after a few minutes without resuming. The log is from a session where playback didn't start at all. In either case the video does not play when I skip forward or go backward.

**Comment by chinghei1007** [comment]
## Environment

Android version: 14
Custom ROM: No

## Screenshot of passing SafetyNet checks using a SafetyNet checker

![Screenshot_2024-05-03-14-56-38-211_rikka safetynetchecker](https://github.com/ReVanced/revanced-discussions/assets/35035492/4f1e351c-f79a-4305-bcd5-062118f7edc4)

## Screenshot of GmsCore settings, especially the SafetyNet setting

![Screenshot_2024-05-03-14-56-19-520_app revanced android gms](https://github.com/ReVanced/revanced-discussions/assets/35035492/abb32e04-e8c6-4c59-be95-4809e955cad2)

## Logs captured from the troubleshooting section

```05-03 14:50:23.563  9681  9681 D PoTokens: onBind: Intent 
05-03 14:50:23.821  9681 11266 D PoTokens: bound by: GetServiceRequest{serviceId=POTOKENS, gmsVersion=241002000, packageName='app.revanced.android.youtube', extras=Bundle[{}]}
05-03 14:50:23.821  9681 11266 D PoTokens: PoTokensApiService handleServiceRequest
05-03 14:50:23.953  9681  9753 D PoTokens: responseStatusToken packageName: com.google.android.apps.youtube
05-03 14:49:02.599  9681  9681 D PoTokens: responseStatusToken result: 244
05-03 14:49:02.600  9681  9681 D PoTokens: responseStatusToken end
```

i think it is really, really werid, 
I'm having two POCO F5, all in default ROM
on same version of HyperOS

the logs are from the phone 1 (which has problem), i just factory reset it because I just felt like it

so phone 1 having 19.11.43 default patches, and the Patched version doesn't like to play.
Auto quality keeps at 480p but it will play for seconds
Then stuck for minutes to play for another half a minute
it just stopped playing today, i thought it is the battery restriction on the GMSCore, but I tried everything with no success

my phone 2 have an older version of the the patch, 19.09.37 
having identical ticks on the GmsCore self-check
works perfectly fine

my Samsung Tab S8 also running the 19.09.37 of the patch,
also works perfectly fine

so I tried to extract the working patched apks from the two devices
but they still do not play on phone 1.

So I tried the latest two versions of GmsCore (default and huawei/XiaoMi), and the second lastest one with no success. but my two other devices are actually having the newest version (like i don't know where the problem is)
aiyaaaa

**Reply by oSumAtrIX** [reply]
Try using a different YouTube account. Check if that makes any difference

**Comment by cyberboh** [comment]
@chinghei1007 If this caused by A/B testing by Google, try use another account on your GmsCore. 
1. Sign out / remove all accounts from GmsCore 
2. Sign in with another account or create new fresh Google account and then signed in into GmsCore
3. Or if it's still does not works, use VPN before opening YouTube revanced.

**Reply by JoshWoddy** [reply]
creating a new account seems to work for me
FYI : This is tested only for a few minutes and might not work in the future, i will use the new account and see what happen

**Reply by Splatypus** [reply]
Swapping accounts on YouTube worked for me too. Didn't even remove the old one. Just swapped what I was signed into gms/YouTube with. I've only tested the new one for a few minutes, but the problem seems entirely gone.   
Seems to be consistent with multiple accounts working (one new, one old that I just rarely use). Swapping back to the original account immediately runs into this issue again.

**Reply by shakthifuture** [reply]
Same issue i am facing, orignal account not working

**Comment by iDoNtCaRe-69** [comment]
## Environment

Android version: 14 QPR1
Custom ROM: Yes (Derpfest)

## Screenshot of passing SafetyNet checks using a SafetyNet checker

![Screenshot_20240503-204743_Yet_Another_SafetyNet_Attestation_Checker](https://github.com/ReVanced/revanced-discussions/assets/59225664/9084c01b-0c34-45ef-a27b-f408ecbc4b85)

## Screenshot of GmsCore settings, especially the SafetyNet setting

![Screenshot_20240503-204838_microG_Services](https://github.com/ReVanced/revanced-discussions/assets/59225664/e990eebe-f909-499c-8311-3305a52de6d6)

![Screenshot_20240503-204900_microG_Services](https://github.com/ReVanced/revanced-discussions/assets/59225664/26d45bcf-5a57-4cdd-8dfe-62cfb2c5bf0d)

## Logs captured from the troubleshooting section

05-03 20:58:41.451 29242  8502 D ChimeraServiceProvider: serviceIntentCall: app.revanced.android.gms.potokens.service.START -> Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
05-03 20:58:41.459 29242 29242 D PoTokens: onBind: Intent { act=app.revanced.android.gms.potokens.service.START pkg=app.revanced.android.gms cmp=app.revanced.android.gms/org.microg.gms.potokens.PoTokensService }
05-03 20:58:41.883 29242  8502 D PoTokens: bound by: GetServiceRequest{serviceId=POTOKENS, gmsVersion=241702000, packageName='app.rvx.android.youtube', extras=Bundle[{}]}
05-03 20:58:41.883 29242  8502 D PoTokens: PoTokensApiService handleServiceRequest
05-03 20:58:42.059 29242  8502 D PoTokens: responseStatusToken packageName: com.google.android.apps.youtube
05-03 20:58:42.067 29242 29242 D PoTokens: responseStatusToken start
05-03 20:58:42.073 29242 29242 D PoTokens: responseStatusToken result: 245
05-03 20:58:42.073 29242 29242 D PoTokens: responseStatusToken end

## Additional Notes:
There are couple of things i noticed so far with this sudden outbreak,
1. This issue is limited to YouTube Mobile Apps. Tested it on multiple devices, for example Android TV and Laptops and they work fine. 
+ Android TV = [SmarTube](https://github.com/yuliskov/SmartTube)
+ Laptop with Browser Extensions = [Enhancer for YouTube™](https://chromewebstore.google.com/detail/enhancer-for-youtube/ponfpcnoihfmfllpaingbgckeeldkhle)
2. It is happening with some other apps too like SnapTube and different mods of YouTube Revanced.
3. It started happening from May 3rd, similar thing happened in past but most of them had a working fix. This time nothing seems to work except making a new account or using a different one. 
4. (Rumor) I recently heard that YouTube will use some other Anti-adblocking method, and one of them had a screenshot showing "Video can't be Played". I was able to reproduce it by spoofing the version, or "spoof the player parameter" option.

![Screenshot_20240503-125312_YouTube](https://github.com/ReVanced/revanced-discussions/assets/59225664/2da3495f-38d0-4a36-8951-51e48fd2b843)

**Comment by oSumAtrIX** [comment]
~~Since there's multiple reports now, I'll need access to an account to debug the issue. Reach out to osumatrix@revanced.app for further instructions, no account credentials are necessary~~

I have an account to debug now
